Historically, <sys/mman.h> did not define madvise() properly for C++,
leading libtorrent to compensate by defining it directly in the code.
Since then, <sys/mman.h> has been fixed. Revert to the header, because
it is the right thing to do, and because it allows us to get across the
transition from the obsolete caddr_t definition to the modern (void *). See:

	29897679 ON headers use caddr_t for standard interfaces
		that should be (void *)

--- libtorrent-0.13.6/scripts/common.m4.orig	2016-09-15 17:07:33.144629366 +0000
+++ libtorrent-0.13.6/scripts/common.m4	2016-09-15 17:08:17.259446458 +0000
@@ -171,7 +171,7 @@
   AC_COMPILE_IFELSE([AC_LANG_SOURCE([
       #include <sys/types.h>
       #include <sys/mman.h>
-        void f() { static char test@<:@1024@:>@; madvise((void *)test, sizeof(test), MADV_NORMAL); }
+        void f() { static char test@<:@1024@:>@; madvise((char *)test, sizeof(test), MADV_NORMAL); }
       ])],
     [
       AC_MSG_RESULT(yes)
--- libtorrent-0.13.6/src/data/memory_chunk.cc.orig	2019-06-15 03:49:32.429413467 +0000
+++ libtorrent-0.13.6/src/data/memory_chunk.cc	2019-06-15 03:50:00.246042180 +0000
@@ -45,11 +45,7 @@
 #include "memory_chunk.h"
 
 #ifdef __sun__
-extern "C" int madvise(void *, size_t, int);
-//#include <sys/mman.h>
-//Should be the include line instead, but Solaris
-//has an annoying bug wherein it doesn't declare
-//madvise for C++.
+#include <sys/mman.h>
 #endif
 
 namespace torrent {
